### PR TITLE
Utility::random() and srandom() is not used anymore.

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Eliminate rand() and srand() usage. These are really bad random generators and should not be used. Leaving usage of random() alone.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/common_startup.cc [hunk 1]
```diff
@@ -496,9 +496,7 @@ static void triggerLoadOfLibraries()
 
 void mainthread()
 {
-   Utility::srandom(time(0) ^ getpid());
```
Comment (rgacogne) [inline pdns/common_startup.cc]: It looks like `random()` is still used in `waitForMultiData()`, `waitFor2Data()` (this one seems unused, though) and `Lua` records, so perhaps we still need to seed it properly?

Comment (omoerbeek) [reply pdns/common_startup.cc]: Yes, looking into this problem as we speak. e.g. pdns_notify uses random() for it's DNS id and always sends the same one.

**Comment by omoerbeek** [comment]
The calls of random() in misc are a bit hard to avoid by replacing them with dns_random(), since the would introduce dependencies on libcrypto and possibly libsodium in a lot of utils. Don't know if that is worth the trouble.
AFAIKS, the non-seeded use of random() are not a big issue in misc.cc.

**Comment by pieterlexis** [comment]
> The calls of random() in misc are a bit hard to avoid by replacing them with dns_random(), since the would introduce dependencies on libcrypto and possibly libsodium in a lot of utils. Don't know if that is worth the trouble.

libcrypto is already a dependency for the auth and the recursor, I don't think it is a problem to have several of our tools depend on it as well.

**Comment by omoerbeek** [comment]
If dependencies on libcrypto (and libsodium) are not considered a problem, I might prepare a bigger hammer to move everything to dns_random and eliminate random() everywhere. A kind of preventive measure to avoid having a case where predictable random is actually problem while we thought it was not.

**Comment by rgacogne** [comment]
libcrypto is already required in the auth and rec, but not in dnsdist, so please be careful about that.

**Comment by omoerbeek** [comment]
Looks like dnsdist already has a (conditional) dependency on libcrypto if HAVE_DNS_OVER_TLS && HAVE_LIBSSL

**Comment by rgacogne** [comment]
Indeed, but I really would like that dependency to remain optional.

**Comment by omoerbeek** [comment]
I'm parking this for the moment. It introduces a strange error in the rec testrunner on travis I cannot reproduce on any system available to me. I need to reproduce the travis environment exactly, it seems.

**Comment by omoerbeek** [comment]
Failure on travis tracked and fixed. Should be good to merge.

**Pull request merged**